src/search.ts

```diff
@@ -1,11 +1,15 @@
 import * as findInFiles from 'find-in-files';
 
 const regExprRequiredVersion = /(?<=(required_version.=.)).*/;
-const regExprSubDirectory = /(wrappers|examples|modules)/;
+// We only want files at the current directory level, exlcude sub-directories
+const regExprSubDirectory = /\//;
 
 export async function versionConstraintSearch(dir: string): Promise<string> {
   const files = await findInFiles.find('required_versions*s*', dir, '.tf$');
-  const key = Object.keys(files).filter(word => !regExprSubDirectory.test(word))[0];
+
+  const key = Object.keys(files)
+    .sort((a, b) => a.length - b.length)
+    .filter(word => !regExprSubDirectory.test(word))[0];
```

Comment:
I can see you're still fighting this :sweat: 
By denying the regex you are removing any case that is not from the root path of the checkout, that is from the project root and not from the sub path. We must take into account that findInFiles returns the path from the root like `tests/0.13/versions.tf`, working with array like this: `["tests/0.13/versions.tf","tests/0.12/versions.tf","tests/terraform-aws-alb/versions.tf","tests/terraform-aws-alb/wrappers/versions.tf","tests/terraform-aws-alb/examples/complete-nlb/versions.tf","tests/terraform-aws-alb/examples/complete-alb/versions.tf"]`.
With this negation we do not get any file.

I have [this test adding](https://github.com/jotasixto/terraform-aws-alb/pull/1) the skip of the wrappers dir in the target workflow, but the truth is not very elegant.

Another option is to add working directory logic to this piece of code, to run from the root of the desired final directory.

Reply:
ah yes, I can't ignore all sub-directories, just paths with `wrappers` in it. thanks

Reply:
I don't even know if I need the regexp filtering, I think I see where you're coming from with the length check. But everything seems to be jiving now and we're doubly ensured we won't evaluate `wrappers/` 😅 - thanks a ton for your help and watching as I struggle through this!!!